BHoM_UI/Templates/MethodCaller.cs

```diff
@@ -238,11 +173,28 @@ public virtual void SetInputParams()
                     HasDefaultValue = x.HasDefaultValue,
                     DefaultValue = x.DefaultValue
                 }).ToList();
+                if (Method is MethodInfo && !Method.IsStatic)
+                {
+                    InputParams.Insert(0, new ParamInfo
+                    {
+                        Name = Method.DeclaringType.Name.ToLower(),
+                        DataType = Method.DeclaringType,
+                        Description = "",
+                        Kind = ParamKind.Input,
+                        HasDefaultValue = false,
+                        DefaultValue = System.DBNull.Value
+                    });
+                }
             }
         }
 
         /*************************************/
 
+        public virtual void SetDelegate()
+        {
+            m_CompiledFunc = Method.ToFunc();
+        }
+
         public virtual void SetOutputParams()
```

Comment:
You forgot to put the separator comment here.
Although to be honest, I am not sure I see the need of this method as it is a one liner is is only called once